module-bsp/board/rt1051/bsp/magnetometer/magnetometer.cpp

```diff
@@ -118,234 +122,280 @@ namespace bsp
-            reg_int.int_X_en             = als31300::INT_REG_INT_CHANNEL_disabled;
-            reg_int.int_Y_en             = als31300::INT_REG_INT_CHANNEL_disabled;
-            reg_int.int_Z_en             = als31300::INT_REG_INT_CHANNEL_disabled;
-            reg_int.int_X_threshold      = 1;
-            reg_int.int_Y_threshold      = 4;
-            reg_int.int_Z_threshold      = 0;
-            if (current_reg_int != reg_int) {
-                [[maybe_unused]] auto ret = i2cWrite(als31300::INT_REG, reg_int);
-                assert(ret);
-                LOG_DEBUG("INT wrote:\t%" PRIu32, static_cast<uint32_t>(reg_int));
-
-                i2cRead(als31300::INT_REG, read_reg);
-                LOG_DEBUG("INT verify:\t%" PRIu32, static_cast<uint32_t>(als31300::int_reg(read_reg)));
+            if (i2cRead(als31300::CONF_REG, readRegister)) {
+                LOG_DEBUG("CONF register value verification: %" PRIu32,
+                          static_cast<std::uint32_t>(als31300::conf_reg(readRegister)));
             }
             else {
-                LOG_DEBUG("INT is fine, sparing a write");
-            }
-
-            if (timerHandle == nullptr) {
-                timerHandle =
-                    xTimerCreate("SliderTimer", pdMS_TO_TICKS(magnetometerPollIntervalMs), true, nullptr, TimerHandler);
-                if (timerHandle == nullptr) {
-                    LOG_ERROR("Could not create the timer for magnetometer state change detection!");
-                    return kStatus_Fail;
-                }
+                LOG_ERROR("Failed to read from CONF register!");
             }
+        }
+        else {
+            LOG_DEBUG("CONF register unchanged, skipping write");
+        }
 
-            // POWER register
-            i2cRead(als31300::PWR_REG, read_reg);
-            const als31300::pwr_reg current_reg_pwr = read_reg;
-            LOG_DEBUG("POWER read:\t%" PRIu32, static_cast<uint32_t>(current_reg_pwr));
-            als31300::pwr_reg reg_pwr = current_reg_pwr;
-            reg_pwr.I2C_loop_mode     = als31300::PWR_REG_LOOP_MODE_single; // we don't want constant data flow
-            reg_pwr.sleep             = als31300::PWR_REG_SLEEP_MODE_active;
-            reg_pwr.count_max_LP_mode = static_cast<uint8_t>(LPDCM_INACTIVE_TIME::inactive_10ms);
-
-            i2cWrite(als31300::PWR_REG, reg_pwr);
-            LOG_DEBUG("POWER wrote:\t%" PRIu32, static_cast<uint32_t>(reg_pwr));
+        // INTERRUPTS register
+        readStatus                                 = i2cRead(als31300::INT_REG, readRegister);
+        const als31300::int_reg currentIntRegister = readRegister;
+        if (readStatus) {
+            LOG_DEBUG("Current INT register value: %" PRIu32, static_cast<std::uint32_t>(currentIntRegister));
+        }
+        else {
+            LOG_ERROR("Failed to read INT register!");
+        }
 
-            xTimerStart(timerHandle, 1000);
+        als31300::int_reg newIntRegister    = currentIntRegister;
```

Comment:
Minor: please avoid `new` in variable names so that it can't be confused with memory allocation e.g. when searching

Reply:
Changed